pkg/api/validation/validation.go

```diff
@@ -1125,12 +1125,9 @@ func validateFinalizerName(stringValue string) errs.ValidationErrorList {
 func ValidateNamespaceUpdate(oldNamespace *api.Namespace, namespace *api.Namespace) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
 	allErrs = append(allErrs, ValidateObjectMetaUpdate(&oldNamespace.ObjectMeta, &namespace.ObjectMeta).Prefix("metadata")...)
-
-	// TODO: move reset function to its own location
-	// Ignore metadata changes now that they have been tested
-	oldNamespace.ObjectMeta = namespace.ObjectMeta
 	oldNamespace.Spec.Finalizers = namespace.Spec.Finalizers
-
+	oldNamespace.Labels = namespace.Labels
+	oldNamespace.Annotations = namespace.Annotations
```

Comment:
this is what happens when you are rushing to fix something before leaving work ;-)

cleaning this up more this morning and getting more tests for various strategies.